helix-vcs/src/lib.rs

```diff
@@ -0,0 +1,47 @@
+use std::{collections::HashMap, path::Path};
+
+pub use differ::Differ;
+pub use git::Git;
+
+mod differ;
+mod git;
+mod rope_line_cache;
+
+// TODO: Move to helix_core once we have a generic diff mode
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum LineDiff {
+    Added,
+    Deleted,
+    Modified,
+}
+
+/// Maps line numbers to changes
+pub type LineDiffs = HashMap<usize, LineDiff>;
+
+trait DiffProvider {
+    fn get_file_head(&self, file: &Path) -> Option<Vec<u8>>;
+}
+pub struct DiffProviderRegistry {
+    providers: Vec<Box<dyn DiffProvider>>,
+}
+
+impl DiffProviderRegistry {
+    pub fn new() -> DiffProviderRegistry {
+        // currently only git is supported
+        // TODO make this configurable when more providers are added
+        let git: Box<dyn DiffProvider> = Box::new(Git);
+        let providers = vec![git];
+        DiffProviderRegistry { providers }
+    }
```

Comment:
Why not remove `new()` and only have `Default`?

Reply:
I only added `default` to make clippy shutup :D I think that keeping new is the better option because I want to make the differs configurable in the future and then some sort of configuration would be passed as an argument to new. But for this PR I can also only implement default if you think that is a better fit

Reply:
I think both is fine, but having only default is better here so we don't have to duplicate it to new?

Reply:
I have removed the `new` function in the latest commit